packages/management/tkg-autoscaler/bundle/config/upstream/tkg-autoscaler.yaml

```diff
@@ -0,0 +1,211 @@
+#@ load("@ytt:data", "data")
+
+#@ def autoscaler_volume_name():
+#@   return "{}-cluster-autoscaler-volume".format(data.values.CLUSTER_NAME)
+#@ end
+
+#@ def autoscaler_kubeconfig_mount_path():
+#@   return "/mnt/{}-kubeconfig".format(data.values.CLUSTER_NAME)
+#@ end
+
+#@ def autoscaler_args():
+#@   args = ["--cloud-provider=clusterapi"]
+#@   args.append("--v=4")
+#@   args.append("--clusterapi-cloud-config-authoritative")
+#@   args.append("--kubeconfig=" + autoscaler_kubeconfig_mount_path() + "/value")
+#@   args.append("--node-group-auto-discovery=clusterapi:clusterName=" + data.values.CLUSTER_NAME)
+#@   args.append("--scale-down-delay-after-add=" + data.values.AUTOSCALER_SCALE_DOWN_DELAY_AFTER_ADD)
+#@   args.append("--scale-down-delay-after-delete=" + data.values.AUTOSCALER_SCALE_DOWN_DELAY_AFTER_DELETE)
+#@   args.append("--scale-down-delay-after-failure=" + data.values.AUTOSCALER_SCALE_DOWN_DELAY_AFTER_FAILURE)
+#@   args.append("--scale-down-unneeded-time=" + data.values.AUTOSCALER_SCALE_DOWN_UNNEEDED_TIME)
+#@   args.append("--max-node-provision-time=" + data.values.AUTOSCALER_MAX_NODE_PROVISION_TIME)
+#@   args.append("--max-nodes-total=" + "{}".format(data.values.AUTOSCALER_MAX_NODES_TOTAL))
+#@   return args
+#@ end
+
+#@ if data.values.TKG_CLUSTER_ROLE == "workload" and data.values.PROVIDER_TYPE != "tkg-service-vsphere":
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: #@ "{}-cluster-autoscaler".format(data.values.CLUSTER_NAME)
+  namespace: #@ data.values.NAMESPACE
+  labels:
+    app: #@ "{}-cluster-autoscaler".format(data.values.CLUSTER_NAME)
+spec:
+  selector:
+    matchLabels:
+      app: #@ "{}-cluster-autoscaler".format(data.values.CLUSTER_NAME)
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: #@ "{}-cluster-autoscaler".format(data.values.CLUSTER_NAME)
+    spec:
+      containers:
+        - image: tkg-autoscaler:latest
```

Comment:
That can't be right. An unexpected pod restart shouldn't result in a change to the autoscale image.

Reply:
This is going to get replaced with a fixed version tag when we're building the package using `kbld`